insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go

```diff
@@ -644,68 +636,61 @@ func TestValidationInspector_ActiveClusterIdsNotSet_Prune_Detection(t *testing.T
+	// override the topic ID for each PRUNE to avoid duplicate topic ID errors
+	ctlMsgs := spammer.GenerateCtlMessages(1, p2ptest.WithPrune(10, clusterPrefixedTopic.String()))
+	for _, g := range ctlMsgs[0].GetPrune() {
+		s := randomClusterPrefixedTopic().String()
+		g.TopicID = &s
+	}
 	// start spamming the victim peer
 	spammer.SpamControlMessage(t, victimNode, ctlMsgs)
 
 	unittest.RequireCloseBefore(t, done, 5*time.Second, "failed to inspect RPC messages on time")
 }
 
 // TestValidationInspector_Unstaked_Node_Detection ensures that RPC control message inspector disseminates an invalid control message notification when an unstaked peer
-// sends a control message for a cluster prefixed topic.
+// sends an RPC.
 func TestValidationInspector_UnstakedNode_Detection(t *testing.T) {
-	// TODO: we must implement checking rpcs for unstaked peers right at the beginning of the inspection process.
-	unittest.SkipUnless(t, unittest.TEST_TODO, "the feature this test is testing is not yet implemented")
 	role := flow.RoleConsensus
 	sporkID := unittest.IdentifierFixture()
 	flowConfig, err := config.DefaultConfig()
 	require.NoError(t, err)
 	inspectorConfig := flowConfig.NetworkConfig.GossipSub.RpcInspector.Validation
-	// set hard threshold to 0 so that in the case of invalid cluster ID
-	// we force the inspector to return an error
-	inspectorConfig.ClusterPrefixedMessage.HardThreshold = 0
 	inspectorConfig.InspectionQueue.NumberOfWorkers = 1
-
-	// SafetyThreshold < messageCount < HardThreshold ensures that the RPC message will be further inspected and topic IDs will be checked
-	// restricting the message count to 1 allows us to only aggregate a single error when the error is logged in the inspector.
-	messageCount := 10
 	controlMessageCount := int64(1)
 
 	count := atomic.NewInt64(0)
 	done := make(chan struct{})
-	expectedNumOfLogs := 2
-	hook := zerolog.HookFunc(func(e *zerolog.Event, level zerolog.Level, message string) {
-		if level == zerolog.WarnLevel {
-			if message == "control message received from unstaked peer" {
-				count.Inc()
-			}
-		}
-		if count.Load() == int64(expectedNumOfLogs) {
-			close(done)
-		}
-	})
-	logger := zerolog.New(os.Stdout).Level(zerolog.WarnLevel).Hook(hook)
 
 	idProvider := mock.NewIdentityProvider(t)
+	inspectorIDProvider := mock.NewIdentityProvider(t)
 	spammer := corruptlibp2p.NewGossipSubRouterSpammer(t, sporkID, role, idProvider)
 	ctx, cancel := context.WithCancel(context.Background())
 	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
 
 	consumer := mockp2p.NewGossipSubInvalidControlMessageNotificationConsumer(t)
 	consumer.On("OnInvalidControlMessageNotification", mockery.Anything).Run(func(args mockery.Arguments) {
-		// TODO: here we should assert the notification received
-	}).Return().Once()
+		count.Inc()
+		notification, ok := args[0].(*p2p.InvCtrlMsgNotif)
+		fmt.Println(notification.Error)
```

Comment:
I guess we don't need this.

Reply:
https://github.com/onflow/flow-go/pull/5449/commits/7482fcf826a90cf7026aa6c2ad74f7415010826f